### PR TITLE
Get all pod logs on Kube CI failure

### PR DESCRIPTION
**Pull request by dimberman**
uses `--all-containers=true` flag to get all k8s logs

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by ashb** [comment]
@dimberman Can you also make it dump the logs/status of postgres, and I guess kubedns/coredns/whatever DNS is actually in use?

**Comment by dimberman** [comment]
@ashb yes was just thinking that

File: scripts/ci/libraries/_kind.sh [hunk 1]
```diff
@@ -365,19 +365,23 @@ function apply_kubernetes_resources() {
...
-    kubectl logs "${POD}" -c scheduler --cluster "${KUBECTL_CLUSTER_NAME}" || true
-    echo "--------------"
+    echo "------- airflow pod logs -------"
+    kubectl logs "${POD}" --all-containers=true || true
+    PG_POD=POD=$(kubectl get pods -o go-template --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}' \
+        --cluster "${KUBECTL_CLUSTER_NAME}" | grep postgres | head -1)
+    echo "------- postgres pod description -------"
+    kubectl describe pod "${PG_POD}" --cluster "${KUBECTL_CLUSTER_NAME}"
+    echo "------- postgres pod logs -------"
+    kubectl logs "${PG_POD}" --all-containers=true || true
```
Comment (ashb) [inline scripts/ci/libraries/_kind.sh]: Something not quite right here: 
```
------- postgres pod description -------
Error from server (NotFound): pods "POD=postgres-airflow-685c8f6684-6nxnl" not found
###########################################################################################
                   EXITING WITH STATUS CODE 1
###########################################################################################
```

File: scripts/ci/libraries/_kind.sh [hunk 2]
```diff
@@ -365,19 +365,23 @@ function apply_kubernetes_resources() {
...
-        kubectl logs "${POD}" -c git-sync-clone --cluster "${KUBECTL_CLUSTER_NAME}" || true
-    fi
-    echo "------- webserver logs -------"
-    kubectl logs "${POD}" -c webserver --cluster "${KUBECTL_CLUSTER_NAME}" || true
-    echo "------- scheduler logs -------"
-    kubectl logs "${POD}" -c scheduler --cluster "${KUBECTL_CLUSTER_NAME}" || true
-    echo "--------------"
+    echo "------- airflow pod logs -------"
+    kubectl logs "${POD}" --all-containers=true || true
+    PG_POD=POD=$(kubectl get pods -o go-template --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}' \
```
Comment (ashb) [inline scripts/ci/libraries/_kind.sh]:
```suggestion
    PG_POD=$(kubectl get pods -o go-template --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}' \
```

File: scripts/ci/libraries/_kind.sh [hunk 3]
```diff
@@ -365,19 +365,23 @@ function apply_kubernetes_resources() {
 function dump_kubernetes_logs() {
     POD=$(kubectl get pods -o go-template --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}' \
         --cluster "${KUBECTL_CLUSTER_NAME}" | grep airflow | head -1)
+    kubectl get all
+    kubectl get all --namespace kube-system
+    echo "------coredns logs-------"
+    DNS_POD=$(kubectl get pods -o go-template -n kube-system --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}' \
+        --cluster "${KUBECTL_CLUSTER_NAME}" | grep dns | head -1)
```
Comment (ashb) [inline scripts/ci/libraries/_kind.sh]: You forgot the logs command :dagger: 

Also an easier way for this

```suggestion
  kubectl -n kube-system logs -l k8s-app=kube-dns --all-containers --cluster "${KUBECTL_CLUSTER_NAME}"
```

(Even with coredns, the k8s-app label is still kube-dns)

**Comment by potiuk** [comment]
Hey @dimberman @ashb  ->  I see there were some problems with Kubernetes on CI . I tried to reproduce it locally https://github.com/apache/airflow/blob/master/TESTING.rst#typical-testing-pattern-for-kubernetes-tests but of course app can be deployed there without any problems. 

I have started long weekend and have little access to computer now - so I hope you will solve it eventually but for what it can be worth - we already had a function that was sending all the logs from kind (kind has the useful "kind export logs" feature for that  but it was not wired to the scripts after the recent refactor. I added it to send logs always for now in #9228 .

I planned to use Github Action artifacts instead of file.io - so maybe you can do just that as well.

**Comment by potiuk** [comment]
BTW. Did you figure out why it suddenly stopped working - I see that you modified coredns configuration again - something that we've done long time ago when kind was starting from the container

**Comment by potiuk** [comment]
Can I help somehow ? I have a little time before I get to my car in 400 km trip :(

**Review by potiuk — APPROVED** [review]
Looks good (for now) with the coredns patch but the comments should be removed and #9228  should be used instead to get all the logs from kubernetes

**Comment by ashb** [comment]
Not needed now for #9230, but using this to investigate the git init problems.

**Comment by ashb** [comment]
Giving up for now - not getting anywhere with why the git-sync tests are failing.

**Pull request closed without merging**